src/tools/std23/views/repeat.hpp

```diff
@@ -0,0 +1,235 @@
+          }
+          friend constexpr auto operator< (const iterator& left, const iterator& right) {
+              return right.current_ > left.current_;
+          }
+
+          friend constexpr iterator operator+(iterator i, difference_type n) {
+              i += n;
+              return i;
+          }
+
+          friend constexpr iterator operator+(difference_type n, iterator i) {
+              i += n;
+              return i;
+          }
+
+          friend constexpr iterator operator-(iterator i, difference_type n) {
+              i -= n;
+              return i;
+          }
+
+          friend constexpr difference_type operator-(const iterator& left, const iterator& right) {
+              return static_cast<difference_type>(left.current_) - static_cast<difference_type>(right.current_);
+          }
+
+          
+    };
+
+public:
+
+    repeat_view() = default;
+
+    constexpr explicit repeat_view( const Type& value, Bound bound_sentinel = Bound())
+        : value_(std::in_place, value), bound_(bound_sentinel) {
+        }
+
+    constexpr explicit repeat_view(Type&& value, Bound bound_sentinel = Bound())
+        : value_(std::in_place, std::move(value)), bound_(bound_sentinel) {
+    }
+
+    template <class... _TpArgs, class... _BoundArgs, 
+          std::enable_if_t<std20::constructible_from<Type, _TpArgs...> && std20::constructible_from<Bound, _BoundArgs...>, bool> = true>
+    constexpr explicit repeat_view( std::piecewise_construct_t, std::tuple<_TpArgs...> __value_args, 
+                    std::tuple<_BoundArgs...> __bound_args = std::tuple<>{}) 
+        : value_(std::in_place, std::make_from_tuple<Type>(std::move(__value_args))),
+        bound_(std::make_from_tuple<Bound>(std::move(__bound_args))) {
+    }
+
+    constexpr iterator begin() const { return iterator(std::addressof(*value_)); }
+
+    template<typename T=Bound, std::enable_if_t<!std20::same_as<T, std20::unreachable_sentinel_t>, bool> = true> 
+    constexpr iterator end() const
+    {
+        return iterator(std::addressof(*value_), bound_);
+    }
+
+    template<typename T=Bound, std::enable_if_t<std20::same_as<T, std20::unreachable_sentinel_t>, bool> = true> 
+    constexpr std20::unreachable_sentinel_t end() const noexcept 
+    {
+        return std20::unreachable_sentinel; 
+    }
+
+    template<typename T=Bound, std::enable_if_t<!std20::same_as<T, std20::unreachable_sentinel_t>, bool> = true> 
+    constexpr auto size() const
+    {
+	return std::make_unsigned_t<decltype(bound_)>(bound_);
+    }
+};
+
+template < typename Type >
+repeat_view( Type&&, std20::ranges::range_difference_t< Type > ) -> repeat_view< std20::ranges::all_view< Type > >;
+
+} // namespace ranges
+} // namespace std23
+
+namespace std20 {
+inline namespace ranges {
+
+template <typename Type>
+inline constexpr bool enable_borrowed_range<std23::ranges::repeat_view<Type>> = forward_range<Type> && enable_borrowed_range<Type>;
```

Comment:
I don't think this is needed in the case of repeat. For a range adapter like drop which modifies an existing range, it is required though.

Reply:
Done. No issues seen when removing this